lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java

```diff
@@ -402,26 +403,41 @@ public static LottieTask<LottieComposition> fromJsonString(final String json, @N
   @WorkerThread
   public static LottieResult<LottieComposition> fromJsonStringSync(String json, @Nullable String cacheKey) {
     ByteArrayInputStream stream = new ByteArrayInputStream(json.getBytes());
-    return fromJsonReaderSync(JsonReader.of(buffer(source(stream))), cacheKey);
+    return fromSourceSync(source(stream), cacheKey);
+  }
+
+  public static LottieTask<LottieComposition> fromSource(final Source source, @Nullable final String cacheKey) {
```

Comment:
What do you think about naming this `fromJsonSource` to make it clear that this should be a json file, not zip, gzip, etc?

Reply:
I like it. I've tried to see whether I could do the magic bytes check just like you do in other methods but with that Context parameter it gets awkward which is needed for fonts in case they are embedded.